explorer/websockethandlers.go

```diff
@@ -59,55 +60,100 @@ func (exp *explorerUI) RootWebsocket(w http.ResponseWriter, r *http.Request) {
+					log.Debug("Request size over limit")
+					webData.Message = "Request too large"
+					return
+				}
+
 				switch msg.EventId {
 				case "decodetx":
-					webData.EventId = msg.EventId + "Resp"
-					if len(msg.Message) > requestLimit {
-						log.Debug("Request size over limit")
-						webData.Message = "Request too large"
-						break
-					}
 					log.Debugf("Received decodetx signal for hex: %.40s...", msg.Message)
 					tx, err := exp.blockData.DecodeRawTransaction(msg.Message)
 					if err == nil {
 						message, err := json.MarshalIndent(tx, "", "    ")
 						if err != nil {
 							log.Warn("Invalid JSON message: ", err)
-							webData.Message = fmt.Sprintf("Error: Could not encode JSON message")
+							webData.Message = "Error: Could not encode JSON message"
 							break
 						}
 						webData.Message = string(message)
 					} else {
 						log.Debugf("Could not decode raw tx")
 						webData.Message = fmt.Sprintf("Error: %v", err)
 					}
+
 				case "sendtx":
-					webData.EventId = msg.EventId + "Resp"
-					if len(msg.Message) > requestLimit {
-						log.Debugf("Request size over limit")
-						webData.Message = "Request too large"
-						break
-					}
 					log.Debugf("Received sendtx signal for hex: %.40s...", msg.Message)
 					txid, err := exp.blockData.SendRawTransaction(msg.Message)
 					if err != nil {
 						webData.Message = fmt.Sprintf("Error: %v", err)
 					} else {
 						webData.Message = fmt.Sprintf("Transaction sent: %s", txid)
 					}
-				case "getmempooltxs":
-					webData.EventId = msg.EventId + "Resp"
 
+				case "getmempooltxs":
 					exp.MempoolData.Lock()
 					msg, err := json.Marshal(exp.MempoolData)
 					exp.MempoolData.Unlock()
 
 					if err != nil {
 						log.Warn("Invalid JSON message: ", err)
-						webData.Message = fmt.Sprintf("Error: Could not encode JSON message")
+						webData.Message = "Error: Could not encode JSON message"
 						break
 					}
 					webData.Message = string(msg)
+
+				case "getticketpooldata":
+					// Ensures that multiple requests for the same update are
+					// treated as the single request.
+					if !tpEvent.Active {
+						log.Warn("ticketpool event has not updated data.")
+						break
+					}
+
+					tpEvent.Active = false
+
+					cData, gData, err := exp.explorerSource.TicketPoolVisualization(msg.Message)
+					if err != nil {
+						log.Errorf("TicketPoolVisualization error: %v", err)
+						webData.Message = "Error: Failed to successfully fetch ticketpool data"
+						break
+					}
+
+					exp.MempoolData.Lock()
+					var mpData = exp.MempoolData
+					exp.MempoolData.Unlock()
+
+					var mp = dbtypes.PoolTicketsData{}
+
+					if len(mpData.Tickets) > 0 {
+						mp.Time = append(mp.Time, uint64(mpData.Tickets[0].Time))
+						mp.Price = append(mp.Price, mpData.Tickets[0].TotalOut)
+						mp.Mempool = append(mp.Mempool, uint64(len(mpData.Tickets)))
+					} else {
+						log.Warn("No tickets exists in the mempool")
```

Comment:
This is not unexpected, so `log.Debug` (or `Trace`).

wording: `"No tickets exist in mempool"`

Reply:
ok. Noted